zfs-recompress.py

```diff
@@ -135,30 +129,62 @@ def display_thread(qin: queue.SimpleQueue):
             # print every 10th file and anything larger than 20 MiB
             display_name = truncate_filename(filename, 24)
             clear_line()
-            print(f"Processed {count}: {display_name:<24} ({format_size(size):>10} / {format_size(total_size):>10})", end="", flush=True)
+            print(
+                f"Processed {count}: {display_name:<24} ({format_size(size):>10} / {format_size(total_size):>10})",
+                end="",
+                flush=True,
+            )
     clear_line()
     print(f"Processed {count} files, {format_size(total_size)} total")
 
-def get_files(path):
+
+def get_files(path: str):
     for p in pathlib.Path(path).glob("**/*"):
         yield str(p)
 
 
+def parse_args():
+    # cli arguments
+    parser = argparse.ArgumentParser(
+        description="Rewrite of [gary17/zfs-recompress](https://github.com/gary17/zfs-recompress) in Python for better performance"
```

Comment:
I'm not sure if Markdown link is the best format in console output. I'd just drop the link text and wrap link in `<>` so in most supported terminals the user can just Ctrl+click the link.